p2p/net/swarm/swarm.go

```diff
@@ -82,6 +90,8 @@ func NewSwarm(ctx context.Context, listenAddrs []ma.Multiaddr,
 	s.cg.SetTeardown(s.teardown)
 	s.SetConnHandler(nil) // make sure to setup our own conn handler.
 
+	prom.MustRegisterOrGet(peersTotal)
```

Comment:
oh now i understand. sorry, i missed this earlier. this is getting initialized on `init`. I believe this needs to be on the swarm itself (i.e. a gauge per swarm). in many circumstances we create multiple swarms per process.